File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {

        int maxVolume = INT_MIN;
        int i = 0;
        int j = height.size()-1;
        while(i < j)
        {
            int area = (j-i)*min(height[i],height[j]);
            maxVolume = max(maxVolume,area);
            if(height[i] < height[j])
            {
                i++;
            }
            else if(height[i] > height[j])
            {
                j--;
            }
            else
            {
                i++;
                j--;
            }
        }
        return maxVolume;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        max_area = -float('inf')
        while left <= right:
            area = (right-left)*(min(height[left],height[right]))
            if area > max_area:
                max_area = area
            if height[left] < height[right]:
                left += 1
            elif height[left] > height[right]:
                right -= 1
            else:
                left += 1
                right -= 1
        return max_area
        